### PR TITLE
WIP [DO NOT MERGE] on #123, supporting build variants

### PR DESCRIPTION
**Pull request by stkent**
@btkelly here's a first attempt at adding support for build variants. Stuff basically works! There are some interesting challenges remaining, outlined below - let's talk through them next week.

The new tasks generated for the example project are shown below:

screen shot 2016-10-15 at 9 54 45 pm

The global `gnagCheck` task straight-up just depends on all the variant tasks right now. Because of how we're marking the build status in the per-variant versions of this task, execution stops as soon as gnag detects a violation for _any_ of the build types it's running on. Not sure if this is desirable behavior? If not, I have a rough workaround in mind that will involve checking if any other `gnagCheck*` tasks are present in the graph, but it's not obvious to me exactly _when_ we'd want to fail the build...

In that case, we'd also need to disambiguate output from the variant tasks. In fact, we may want to do this anyway to make it clear which variant produced a failure. For example, we'd probably want our output to be transformed from

``` text
checkstyle_report.xml
findbugs.xml
github-markdown.css
gnag.html
pmd.xml
```

to

``` text
checkstyle_report_debug.xml
checkstyle_report_release.xml
findbugs_debug.xml
findbugs_release.xml
github-markdown.css
gnag_debug.html
gnag_release.html
pmd_debug.xml
pmd_release.xml
```

and corresponding log messages to be updated, etc.

Feedback welcome, as always!

bitmoji

### CODE REVIEW

File: version.gradle [hunk 1]
```diff
@@ -1 +1 @@
-project.ext.gnagPluginVersion = '1.2.1'
+project.ext.gnagPluginVersion = '1.3.0-LOCAL'
```
Comment (stkent) [inline version.gradle]: Oops, should revert.

**Comment by stkent** [comment]
Current folder structure:

screen shot 2016-10-27 at 12 49 48 pm

Per discussion, first step would be to create per-variant folders inside the gnag directory to house per-variant reports (names unchanged).

**Comment by stkent** [comment]
Per discussion, we should aim to have composite tasks run _all_ subtasks before marking the build as a success or failure.

**Comment by stkent** [comment]
I'm going to close this PR but keep the branch around in the repo as reference...

**Comment by btkelly** [comment]
:sob:

**Pull request closed without merging**